src/PrestaShopBundle/Form/Admin/ShopParameters/General/PreferencesType.php

```diff
@@ -0,0 +1,197 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Form\Admin\ShopParameters\General;
+
+use PrestaShop\PrestaShop\Adapter\Entity\Order;
+use PrestaShopBundle\Form\Admin\Type\CommonAbstractType;
+use Symfony\Component\Form\Extension\Core\Type\IntegerType;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+
+/**
+ * Class returning the content of the form in the maintenance page.
+ * To be found in Configure > Shop parameters > General > Maintenance
+ */
+class PreferencesType extends CommonAbstractType
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        $configuration = $this->getConfiguration();
+
+        if ($options['is_ssl_enabled']) {
+            $builder
+                ->add('enable_ssl', ChoiceType::class, array(
+                    'choices_as_values' => true,
+                    'choices'  => array(
```

Comment:
Note for later: we have a SwitchType that is going to be merged soon => https://github.com/PrestaShop/PrestaShop/pull/8705

So for now, it's good but later I'll update your work and you will have to rely on SwitchType for every "ON/OFF" input :)